### PR TITLE
python3Packages.metawear: init at 1.0.8

### PR DESCRIPTION
**Pull request by stepbrobd**
###### Description of changes

python3Packages.metawear: init at 1.0.8

metawear
- https://pypi.org/project/metawear/
- https://github.com/mbientlab/metawear-sdk-python

dependency [warble](https://github.com/NixOS/nixpkgs/pull/238628)
- https://github.com/NixOS/nixpkgs/pull/238628

```shell
$ nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"
2 packages built:
python310Packages.metawear python311Packages.metawear
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/metawear/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+, warble
+}:
+
+buildPythonPackage rec {
+  pname = "metawear";
+  version = "1.0.8";
+  format = "setuptools";
+
+  enableParallelBuilding = true;
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/metawear/default.nix:19]:
```suggestion
```

**Comment by StepBroBD** [comment]
> Why are they vendoring nlohman json?

Looks like it's a dependency for their sensor firmware

https://github.com/mbientlab/MetaWear-SDK-Cpp/blob/f67699e5c7eb64401a68157f553f9585e3857a6a/src/metawear/dfu/cpp/file_operations.cpp#L10

**Comment by SuperSandro2000** [comment]
> Looks like it's a dependency for their sensor firmware

I was more asking in the direction if we can unvendor that and use ours?

File: pkgs/development/python-modules/metawear/default.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-gNEI6P6GslNd1DzFwCFndVIfUvSTPYollGdqkZhQ4Y8=";
+  };
+
+  nativeBuildInputs = [
+    gcc11
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/metawear/default.nix:24]: Can we avoid this if we use our nlohmann_json?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by StepBroBD** [comment]
Thanks for the review!

> I was more asking in the direction if we can unvendor that and use ours?

Just changed it, gcc11 -> gcc, unvendored their version of nlohmann_json and using nixpkgs's nlohmann_json.

**Comment by StepBroBD** [comment]
On x86_64-linux:
```shell
$ nix run nixpkgs#nixpkgs-review -- pr 238630
2 packages added:
python310Packages.metawear (init at 1.0.8) python311Packages.metawear (init at 1.0.8)
4 packages built:
python310Packages.metawear python310Packages.metawear.dist python311Packages.metawear python311Packages.metawear.dist
```

File: pkgs/development/python-modules/metawear/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  version = "1.0.8";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-gNEI6P6GslNd1DzFwCFndVIfUvSTPYollGdqkZhQ4Y8=";
+  };
+
+  nativeBuildInputs = [
+    gcc
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/metawear/default.nix:25]: gcc should be contained in the stdenv

File: pkgs/development/python-modules/metawear/default.nix [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    bluez
+    nlohmann_json
+  ];
+
+  postPatch = ''
+    rm MetaWear-SDK-Cpp/src/metawear/dfu/cpp/json.hpp
+
+    substituteInPlace MetaWear-SDK-Cpp/src/metawear/dfu/cpp/file_operations.cpp \
+        --replace '#include "json.hpp"' '#include <nlohmann/json.hpp>'
+    cat MetaWear-SDK-Cpp/src/metawear/dfu/cpp/file_operations.cpp
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/metawear/default.nix:40]:
```suggestion
```

File: pkgs/development/python-modules/metawear/default.nix [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  buildInputs = [
+    boost
+    bluez
+    nlohmann_json
+  ];
+
+  postPatch = ''
+    rm MetaWear-SDK-Cpp/src/metawear/dfu/cpp/json.hpp
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/metawear/default.nix:37]:
```suggestion
    # remove vendored nlohmann_json
    rm MetaWear-SDK-Cpp/src/metawear/dfu/cpp/json.hpp
```

File: pkgs/development/python-modules/metawear/default.nix [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  enableParallelBuilding = true;
+
+  pythonImportsCheck = [ "mbientlab" "mbientlab.metawear" ];
+
+  meta = with lib; {
+    description = "Python bindings for the MetaWear C++ SDK by MbientLab";
+    homepage = "https://github.com/mbientlab/metawear-sdk-python";
+    license = with licenses; [ unfree ];
+    maintainers = with maintainers; [ stepbrobd ];
+    platforms = [ "aarch64-linux" "x86_64-linux" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/metawear/default.nix:58]:
```suggestion
    platforms = stdenv.isLinux;
```

**Comment by StepBroBD** [comment]
Fixed, on x86_64-linux:

```shell
$ nix run nixpkgs#nixpkgs-review -- pr 238630
2 packages added:
python310Packages.metawear (init at 1.0.8) python311Packages.metawear (init at 1.0.8)
4 packages built:
python310Packages.metawear python310Packages.metawear.dist python311Packages.metawear python311Packages.metawear.dist
```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.metawear

**Pull request merged**